hyperspy/axes.py

```diff
@@ -1427,57 +1445,103 @@ def _update_max_index(self):
+            except TypeError as e:
+                raise TypeError(
+                    f'The iterpath `{path}` is not an iterable. '
+                    'Ensure it is an iterable like a list, array or generator.'
+                    ) from e
+            try:
+                if not (inspect.isgenerator(path) or type(path) is GeneratorLen):
+                # If iterpath is a generator, then we can't check its first value, have to trust it
+                    first_indices = path[0]
+                    if not isinstance(first_indices, collections.Iterable):
+                        raise TypeError
+                    assert len(first_indices) == self.navigation_dimension
+            except TypeError as e:
+                raise TypeError(
+                    f"Each set of indices in the iterpath should be an iterable, e.g. `(0,)` or `(0,0,0)`. " 
+                    f"The first entry currently looks like: `{first_indices}`, and does not satisfy this requirement."
+                    ) from e
+            except AssertionError as e:
+                raise ValueError(
+                    f"The current iterpath yields indices of length "
+                    f"{len(path)}. It should deliver incides with length "
+                    f"equal to the navigation dimension, which is {self.navigation_dimension}."
+                    ) from e
+            else:
+                self._iterpath = path
+                self._iterpath_generator = iter(self._iterpath)
+
+    def _get_iterpath_size(self, masked_elements=0):
+        "Attempts to get the iterpath size, returning None if it is unknown"
+        if isinstance(self.iterpath, str):
+            # flyback and serpentine have well-defined lengths <- navigation_size
+            maxval = self.navigation_size - masked_elements
+        else:
+            try:
+                maxval = len(self.iterpath)
+                if masked_elements:
+                    # Checking if mask indices exist in the iterpath could take a long time,
+                    # or may not be possible in the case of a generator.
+                    warnings.warn(
+                    ("The progressbar length cannot be estimated when using both custom iterpath and a mask."
+                    "The progressbar may terminate before it appears complete. This can safely be ignored."),
+                    UserWarning
+                    )
+            except TypeError:
+                # progressbar is shown, so user can monitor "iterations per second"
+                # but the length of the bar is unknown
+                maxval = None
+                warnings.warn(
+                    ("The AxesManager `iterpath` is missing the `__len__` method, so does not have a known length. "
+                    "The progressbar will only show run time and iterations per second, but no actual progress indicator."),
+                    UserWarning
+                    )
```

Comment:
If this is intended behaviour, this shouldn't a warning and adding it as info in the logger should be enough?

Reply:
Yes, I agree with that.